### PR TITLE
CATROID-629 Add sensors for text recognition

### PR DESCRIPTION
**Pull request by dardiv**
Add sensors for text recognition.

### Your checklist for this pull request
Please review the [contributing guidelines](https://github.com/Catrobat/Catroid/blob/develop/README.md) and [wiki pages](https://github.com/Catrobat/Catroid/wiki/) of this repository.

- [x] Include the name of the Jira ticket in the PR’s title
- [x] Include a summary of the changes plus the relevant context
- [x] Choose the proper base branch (*develop*)
- [x] Confirm that the changes follow the project’s coding guidelines
- [x] Verify that the changes generate no compiler or linter warnings
- [x] Perform a self-review of the changes
- [x] Verify to commit no other files than the intentionally changed ones
- [x] Include reasonable and readable tests verifying the added or changed behavior
- [x] Confirm that new and existing unit tests pass locally
- [x] Check that the commits’ message style matches the [project’s guideline](https://github.com/Catrobat/Catroid/wiki/Commit-Message-Guidelines)
- [x] Stick to the project’s gitflow workflow
- [x] Verify that your changes do not have any conflicts with the base branch
- [x] After the PR, verify that all CI checks have passed
- [x] Post a message in the *catroid-stage* or *catroid-ide* [Slack channel](https://catrobat.slack.com) and ask for a code reviewer

### CODE REVIEW

**Review by ndrnour — CHANGES_REQUESTED** [review]
remotely reviewed over a slack call, please adopt your code as discussed

File: catroid/src/androidTest/java/org/catrobat/catroid/uiespresso/content/brick/stage/TextDetectionResourceTest.kt [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+import org.junit.Assert
+import org.junit.Assert.assertFalse
+import org.junit.Rule
+import org.junit.Test
+import org.junit.experimental.categories.Category
+
+class TextDetectionResourceTest {
+    private lateinit var formula: Formula
+    private lateinit var lastBrickInScript: ScriptEvaluationGateBrick
+
```
Comment (ndrnour) [inline catroid/src/androidTest/java/org/catrobat/catroid/uiespresso/content/brick/stage/TextDetectionResourceTest.kt:51]: a grant permission rule for the camera is missing in this test and in  `FaceDetectionResourceTest.java`, 
please add the following rule to both files.
`@get:Rule
    var runtimePermissionRule: GrantPermissionRule =
        GrantPermissionRule.grant(Manifest.permission.CAMERA)`

**Comment by ndrnour** [comment]
one last thing, please increase the  `CURRENT_CATROBAT_LANGUAGE_VERSION` in `Constants.java`

**Review by ndrnour — APPROVED** [review]
LGTM

**Comment by wslany** [comment]
Thanks!

**Pull request merged**